application/classes/Ushahidi/Repository/Form/Stats.php

```diff
@@ -291,13 +291,15 @@ public function getResponseRecipients($form_id, $created_after, $created_before)
             $query
             ->where('created', '<=', $created_before);
         }
-        $query = DB::select('contact_id')
+        $query = DB::select([DB::expr('COUNT(contact_id)'), 'total'])
             ->distinct(true)
             ->from([$query,'targeted_posts'])
             ->join('messages', 'INNER')
-            ->on('messages.post_id', '=', 'targeted_posts.id');
-        $results =  $query->execute($this->db);
-        return $this->getCollection($results->as_array());
+            ->on('messages.post_id', '=', 'targeted_posts.id')
+            ->where('messages.direction','=', 'incoming');
```

Comment:
responses by the targeted survey recipients should be filtered with the message direction